### ISSUE TITLE
`reflect.Copy` Incorrect When Source is Array of Len > 8

### ISSUE THREAD

**Issue by ben-krieger** [tinygo-org/tinygo#4554]
Tested from `dev`. Minimal reproduction:

```go
//go:build ignore

package main

import (
	"fmt"
	"reflect"
)

func main() {
	guid := [9]byte{0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09}
	b := make([]byte, 9)

	// Copy builtin works
	copy(b, guid[:])
	fmt.Printf("%x\n", b)

	// Reflect with slices works
	reflect.Copy(reflect.ValueOf(b), reflect.ValueOf(guid[:]))
	fmt.Printf("%x\n", b)

	// Reflect with array doesn't work
	reflect.Copy(reflect.ValueOf(b), reflect.ValueOf(guid))
	fmt.Printf("%x\n", b)
}
```

Result:

```console
$ tinygo run ./test.go
010203040506070809
010203040506070809
303ac73ca07f000002

$ go run ./test.go
010203040506070809
010203040506070809
010203040506070809
```

Not shown in the reproduction: a `[8]byte` array works fine, so the issue seems to only be for lengths > 8.

**Comment by ben-krieger** [comment]
It seems that both `[8]byte` and `[9]byte` do not have the indirect flag set. However, the `[9]byte` _should_ be indirect and isn't, presumably because the data is greater than 64 bits.

(I'm not sure why the >64 bit size is relevant, though, since the field is either a pointer to data or a pointer to a pointer to data).

**Comment by ben-krieger** [comment]
The `ValueOf` implementation never sets the indirect flag. Are there some primitives (other than arrays over a certain size) which should be marked indirect?

```go
func ValueOf(i interface{}) Value {
	typecode, value := decomposeInterface(i)
	return Value{
		typecode: (*rawType)(typecode),
		value:    value,
		flags:    valueFlagExported,
	}
}
```

**Comment by ben-krieger** [comment]
Perhaps a solution would be this?

```diff
diff --git a/src/reflect/value.go b/src/reflect/value.go
index d1f8cb2f..df0031b9 100644
--- a/src/reflect/value.go
+++ b/src/reflect/value.go
@@ -72,11 +72,15 @@ func decomposeInterface(i interface{}) (unsafe.Pointer, unsafe.Pointer)
 
 func ValueOf(i interface{}) Value {
 	typecode, value := decomposeInterface(i)
-	return Value{
+	v := Value{
 		typecode: (*rawType)(typecode),
 		value:    value,
 		flags:    valueFlagExported,
 	}
+	if v.typecode.Kind() == Array && v.typecode.Size() > unsafe.Sizeof(uintptr(0)) {
+		v.flags |= valueFlagIndirect
+	}
+	return v
 }
 
 func (v Value) Interface() interface{} {
```

**Comment by ben-krieger** [comment]
Added fix and test in #4543 @ 05e2f5d4
